crates/rpc/ipc/src/server/rpc_service.rs

```diff
@@ -0,0 +1,138 @@
+//! JSON-RPC service middleware.
+use futures_util::future::BoxFuture;
+use jsonrpsee::{
+    server::{
+        middleware::rpc::{ResponseFuture, RpcServiceT},
+        IdProvider,
+    },
+    types::{error::reject_too_many_subscriptions, ErrorCode, ErrorObject, Request},
+    BoundedSubscriptions, ConnectionDetails, MethodCallback, MethodResponse, MethodSink, Methods,
+    SubscriptionState,
+};
+use std::sync::Arc;
+
+/// JSON-RPC service middleware.
```

Comment:
this is the same as the jsonrpsee type?

can we reuse that instead?

Reply:
Yes, it is totally the same as the [`jsonrpsee::RpcService`](https://github.com/paritytech/jsonrpsee/blob/d3b1b3afa8b94d9f6f938221358d697286a81b9a/server/src/middleware/rpc/layer/rpc_service.rs#L43), but its field is private and the `RpcService::new` function is `pub(crate)`, furthermore the type of its field `cfg` is `RpcServiceCfg` which is `pub(crate)`, so I am afraid I can't reuse it, right?